File: Year 2/Semester 4/Programming and Projection Enironments/Week 10/ProtobuffProjectJavaORM/Domain/src/main/java/gRPCProject/domain/Ticket.java
package gRPCProject.domain;

import java.io.Serializable;

public class Ticket extends Entity<Integer>implements Serializable {
    private Integer idShow;
    private String nameOfCostumer;
    private Integer noOfSeats;

    public Ticket(Integer id, Integer idShow, String nameOfCostumer, Integer noOfSeats) {
        super(id);
        this.idShow = idShow;
        this.nameOfCostumer = nameOfCostumer;
        this.noOfSeats = noOfSeats;
    }

    public Integer getIdShow() {
        return idShow;
    }

    public void setIdShow(Integer idShow) {
        this.idShow = idShow;
    }

    public String getNameOfCostumer() {
        return nameOfCostumer;
    }

    public void setNameOfCostumer(String nameOfCostumer) {
        this.nameOfCostumer = nameOfCostumer;
    }

    public Integer getNoOfSeats() {
        return noOfSeats;
    }

    public void setNoOfSeats(Integer noOfSeats) {
        this.noOfSeats = noOfSeats;
    }

    @Override
    public String toString() {
        return "Ticket{" +
                "id=" + getId() +
                ", idShow=" + idShow +
                ", nameOfCostumer='" + nameOfCostumer +
                ", noOfSeats=" + noOfSeats + '\'' +
                '}';
    }
}

File: Year 2/Semester 4/Programming and Projection Enironments/Week 9/ProtobuffProjectCSharp/model/Ticket.cs
using System;

namespace ProtobuffProject.domain
{
    [Serializable]
    public class Ticket<ID> : Entity<ID>
    {
        private int idShow;
        private string nameOfCostumer;
        private int noOfSeats;

        public Ticket(ID id, int idShow, string nameOfCostumer, int noOfSeats)
        : base(id) {
            this.idShow = idShow;
            this.nameOfCostumer = nameOfCostumer;
            this.noOfSeats = noOfSeats;
        }

        public int IdShow
        {
            get => this.idShow;
            set => this.idShow = value;
        }
        
        public string NameOfCostumer
        {
            get => this.nameOfCostumer;
            set => this.nameOfCostumer = value;
        }
        
        public int NoOfSeats
        {
            get => this.noOfSeats;
            set => this.noOfSeats = value;
        }

        public override string ToString()
        {
            return base.ToString() + " " + idShow + " " + nameOfCostumer + " noOfSeats: " + noOfSeats;
        }
    }
}
